src/api/spec/models/user_spec.rb

```diff
@@ -623,4 +623,48 @@
       end
     end
   end
+
+  describe '#can_modify?' do
+    let(:user) { create(:confirmed_user, :with_home, login: 'hans') }
+    let(:project) { user.home_project }
+
+    context 'when dealing with projects' do
+      context 'when checking against a home project' do
+        it { expect(user.can_modify?(project)).to be true }
+      end
+
+      context 'when checking against a child of a home project' do
+        let(:child_project) { create(:project, name: "#{project.name}:something") }
+
+        before do
+          child_project.relationships.build(user: user,
```

Comment:
Why do you need to do this? There are not relationships involved in your changes right?